src/aura/BGE_BatchGiftEntryTab/BGE_BatchGiftEntryTabHelper.js

```diff
@@ -1,50 +1,75 @@
 ({
+    /******************************** Init Functions *****************************/
+
     doInit: function(component) {
-        var action = component.get('c.getNamespacedListView');
+        var action = component.get('c.getTabModel');
         action.setCallback(this, function (response) {
             const state = response.getState();
             if (state === 'SUCCESS') {
-                let namespacedListView = response.getReturnValue();
-                $A.createComponent(
-                    "lightning:listView",
-                    {
-                        "objectApiName": namespacedListView.objectApiName,
-                        "listName": namespacedListView.listName,
-                        "showActionBar": false,
-                        "enableInlineEdit": false,
-                        "showRowLevelActions": false
-                    },
-                    function (listView, status, errorMessage) {
-                        if (status === "SUCCESS") {
-                            let body = component.get("v.body");
-                            body.push(listView);
-                            component.set("v.body", body);
-                        } else {
-                            this.showToast(component, $A.get('$Label.c.PageMessagesError'), errorMessage, 'error');
-                        }
-                    }
-                );
+                let model = JSON.parse(response.getReturnValue());
+                this.setColumns(component, model.columns);
+                component.set('v.totalNumberOfRows', model.totalNumberOfRows);
+                this.setBatchRows(component, model.batches);
             } else {
                 this.handleApexErrors(component, response.getError());
             }
         });
         $A.enqueueAction(action);
     },
 
+    setColumns: function(component, responseColumns) {
+        responseColumns.forEach(function(col){
+           if (col.type === 'number') {
+               col.cellAttributes = {alignment: "left"};
+           } else if (col.fieldName === 'batchLink') {
+               col.typeAttributes = {label:{fieldName:"Name"},target:"self"};
+           }
+        });
+        component.set('v.batchListColumns', responseColumns);
+    },
+
+    getMoreBatchRows: function(component, event) {
```

Comment:
And description here? Also, thoughts on just finding and setting the isLoading flag instead of passing the event?

Reply:
This is how it was in the docs. We could test it without passing the event. Is there a compelling reason to do it that way? Just less stuff to pass around?